src/Endatix.Infrastructure/Identity/AppUser.cs

```diff
@@ -10,6 +10,16 @@ namespace Endatix.Infrastructure.Identity;
 /// </summary>
 public class AppUser : IdentityUser<long>

```

Comment:
I think we don't need such complications. The implementation now works pretty straightforward and simple.
Of course, it is a matter of opinion and both approaches will work.
However, there are at least a few reasons for keeping it this way:
- It's flat and clearer
- In the AuthService (which is where the validation occurs) we have the clear validation logic and conditions
- Though these are knowledge details, I'm not aware how it will work for persisting the record in the DB. It is quite possible that such changes although looking trivial will lead to some more changes needed for the persistence or the further data utilization
- Using DateTime.UtcNow inside this new class sooner or later will bring a dependency as we'll need to use date/time provider. Of course, this is solvable, but is another complication.

I'm not sure what other points can appear as these just come to my mind at first glance.

My general thinking is that we need to keep the things simple and straightforward. Especially at the current stage of the product, having the mindset of focusing on the problematic areas or changing the implementation when it can really make the code maintainability and extension easier, will help us not only with the implementations but also in reviewing PRs and addressing issues.

Reply:
I suggest you review this article and reconsider your arguments against using value objects - https://ardalis.com/refactoring-value-objects/. Even if you are not willing now, at least I want to win you for the idea. We can always to it later, but I disagree calling value objects complications. I am convinced that that their are the opposite - a simplification over the complexity which is already showing (and that's just the start of it) as the domain model grows. 


| item | without value objects | with value objects |
| ---- | ----- | ------ |
| - It's flat and clearer | once we have 10+ primitive items, this will turn into disadvantage. I see our service classes grow too much and validation logic spread to many places. Primitive Obsession is an anti-pattern. This article does a good job explaining it - https://enterprisecraftsmanship.com/posts/functional-c-primitive-obsession/ | we can express out model better and ecapsulate logic related to it. It's only complex as we haven't done it yet, not because it's complicated imo
| AuthService | In the AuthService (which is where the validation occurs) we have the clear validation logic and conditions | In this case it will be moved to the RefreshToken and meaning in the AuthService will be just to check if it's valid and not over communicate the actual implementation details of why is valid. This is pure encapsulation, so I see it as an improvement
| - Though these are knowledge details, I'm not aware how it will work for persisting the record in the DB. It is quite possible that such changes although looking trivial will lead to some more changes needed for the persistence or the further data utilization | Not change if we keep it this way | With Value  Objects the column names will be prefixed with the Value Object, e.g. `refreshToken_Hash`, so yes migration will do it in the future. This is why I wanted us to do it now
| - Using DateTime.UtcNow inside this new class sooner or later will bring a dependency as we'll need to use date/time provider. Of course, this is solvable, but is another complication. | Agree, but TimeProvider is adding complexity in any case | Time provider will be only needed for Creation logic, so I foresee we will inject it in CommandHandlers or Services that create Value Objects as part of response to a user command |